rbac/default-role-manager/role_manager_test.go

```diff
@@ -295,3 +295,18 @@ func TestMatchingFuncOrder(t *testing.T) {
 	testRole(t, rm, "u1", "g1", true)
 	testRole(t, rm, "u1", "g2", true)
 }
+
+func TestCrossDomainRole(t *testing.T) {
+	rm := NewRoleManager(10)
+
+	_ = rm.AddLink("super_admin", "book_admin", "domain1")
+	_ = rm.AddLink("super_admin", "pen_admin", "domain2")
+	_ = rm.AddLink("alice", "book_admin", "domain2")
+	_ = rm.AddLink("bob", "super_admin", "domain1", "domain2")
```

Comment:
does this line mean that bob is a super admin in domain1 and domain2? based on the tests below I see the opposite

Reply:
No, it means bob in domain1 inherits super_admin in domain2, but bob has not role of super_admin in domain1.

Reply:
@closetool here the role logic is very complicated. I think you can draw a role graph like what I did before and put it as a comment, so everyone can understand it easily:

https://github.com/casbin/casbin/blob/0c7aac93d766aeddea324d7a16fd8be1c700bca5/rbac/default-role-manager/role_manager_test.go#L171-L176